src/client.ts

```diff
@@ -5438,27 +5444,36 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      * @return {Promise} Resolves:
      *    {@link module:models/event-timeline~EventTimeline} timeline with the latest events in the room
      */
-    public async getLatestTimeline(timelineSet: EventTimelineSet): Promise<EventTimeline> {
+    public async getLatestTimeline(timelineSet: EventTimelineSet): Promise<Optional<EventTimeline>> {
```

Comment:
This function is expected to always return an `EventTimeline` and was introduced during a time where `getEventTimeline(...)` always returned a timeline as well, see https://github.com/matrix-org/matrix-js-sdk/pull/2299

This will break the assumptions in `room.refreshLiveTimeline()`

For reference, we're also thinking about how to make `getEventTimeline(...)` play nicely here for another bug scenario in https://github.com/matrix-org/matrix-js-sdk/pull/2521#discussion_r919831536